core/src/main/java/org/codice/imaging/nitf/core/common/NitfDateTime.java

```diff
@@ -80,10 +82,13 @@ public final String getSourceString() {
         This is a best-effort, based on the information available, which can be incomplete. If
         no other information is available, a default Date will be returned.
 
-        @return the value of this object as a Date.
-    */
+        @return the value of this object as a Date, or null if the date is not valid.
+     */
```

Comment:
We should consider dropping this method and just keeping toZonedDateTime() or dropping the whole class and just having the parser return a ZonedDateTime.  While we would lose the sourceString, we won't lose any usable information.